pkg/daemon/cmdreporter/cmdreporter.go

```diff
@@ -0,0 +1,250 @@
+/*
+Copyright 2019 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmdreporter
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+	"syscall"
+
+	"github.com/coreos/pkg/capnslog"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	// AppName is the app name reported by cmd-reporter, notably on the ConfigMap's application label.
+	AppName = "cmd-reporter"
+
+	// ConfigMapStdoutKey defines the key in the ConfigMap where stdout is reported.
+	ConfigMapStdoutKey = "stdout"
+
+	// ConfigMapStderrKey defines the key in the ConfigMap where stderr is reported.
+	ConfigMapStderrKey = "stderr"
+
+	// ConfigMapRetcodeKey defines the key in the ConfigMap where the return code is reported.
+	ConfigMapRetcodeKey = "retcode"
+)
+
+var (
+	logger = capnslog.NewPackageLogger("github.com/rook/rook", "job-reporter-cmd")
+)
+
+// CmdReporter is a process intended to be run in simple Kubernetes jobs. The CmdReporter runs a
+// command in a job and stores the results in a ConfigMap which can be read by the operator.
+type CmdReporter struct {
```

Comment:
For the first implementation of the CmdReporter did you say you'll use it to detect the ceph version? I know this is a WIP PR, just wondering if you're planning to break it out to another PR instead of including it first with the OSD rework?

Reply:
For the first implementation, I was planning to implement the `ceph-volume inventory` reporting.

Reply:
Agreed that it should include something that makes use of it in the same PR. I'm fine if it stays with the OSDs first, just wanted to confirm where you were starting.